### ISSUE TITLE
Warning: unused binding. Linting error with re-frame subscriptions

### ISSUE THREAD

**Issue by n2o** [clj-kondo/clj-kondo#1504]
* clj-kondo v2021.12.16
* macOS / docker linux
* Shell

**problem**

clj-kondo is marking the parameter `subscription` of the example below as an unused binding, which is wrong.

**repro**

```clojure
(require '[re-frame.core :as rf])

(defn- show-id [subscription]     ;; <------ this parameter
  (let [id @(rf/subscribe subscription)]     ;; <------ is here used, but linted as unused
    [:h4 id]))

(defn- ids []
  [show-id [:user/id]])

(rf/reg-sub
 :user/id
 (fn [db _]
   (get-in db [:user :id])))
```

`Warning: src/main/schnaq/interface/user.cljs:3:17: warning: unused binding subscription`

**expected behavior**

This is a false positive, so there should be no warning at all.

**Comment by borkdude** [comment]
Fixed on master.
